bpo-38669: Add check for the type of target (#17034)
python/cpython: Misc/NEWS.d/next/Tests/2019-11-04-02-54-16.bpo-38669.pazXZ8.rst

> Raise TypeError if the target is a string when patching an object.

Reviewer: This can be a bit richer:

```suggestion
Raise :exc:`TypeError` when patching a string with :meth:`unittest.mock.patch.object`.
```

Author: I am not sure we are patching the string in this case. I'll leave it as you wrote it, I guess you know what you are talking about. But in my opinion we are not patching the string here. We are trying to patch an attribute of an object. It would be more clear to say that 
Raise :exc:`TypeError` when the target is a string in :meth:`unittest.mock.patch.object`.

Reviewer: Fair enough – choose whichever you think is better!
